tests/agent_server/test_facts.py

```diff
@@ -270,28 +271,11 @@ async def get_fact(rid, result_code=200, limit=10, lower_limit=2):
 
     await _wait_until_deployment_finishes(client, environment, version)
 
-    await get_fact("test::Fact[agent1,key=key1]")  # undeployable
+    await get_fact("test::Fact[agent1,key=key1]", 404)  # undeployable
     await get_fact("test::Fact[agent1,key=key2]")  # normal
     await get_fact("test::Fact[agent1,key=key3]")  # not present -> present
-    await get_fact("test::Fact[agent1,key=key4]")  # unknown
-    await get_fact("test::Fact[agent1,key=key5]", 503)  # broken
-
-    await agent.stop()
-
-    def wait_until_log_records_are_available() -> bool:
-        try:
-            log_sequence = LogSequence(caplog, allow_errors=False, ignore=["tornado.access"])
-            for i in range(5):
-                log_sequence = log_sequence.contains("inmanta.agent.agent.agent1", logging.ERROR, "Unable to retrieve fact")
-            log_sequence.no_more_errors()
-        except AssertionError:
-            return False
-        else:
-            return True
-
-    # The get_parameter() API calls from the server to the agent are executed asynchronously with respect the
-    # get_param() API calls done from the test case to the server. Here we wait until all log records are available.
-    await retry_limited(wait_until_log_records_are_available, timeout=10)
```

Comment:
I think this is not needed anymore since we return earlier for undeployable resources